dbms/src/Storages/BackgroundProcessingPool.cpp

```diff
@@ -152,9 +152,9 @@ void BackgroundProcessingPool::threadFunction()
         addThreadId(getTid());
     }
 
-    MemoryTracker memory_tracker;
-    memory_tracker.setMetric(CurrentMetrics::MemoryTrackingInBackgroundProcessingPool);
-    current_memory_tracker = &memory_tracker;
+    auto memory_tracker = MemoryTracker::create();
```

Comment:
Is this a test code? since this memTracker did not setNext a root memTracker.

Reply:
This seems not a test code.

Reply:
Root memtracker is total_mem_tracker in ProcessLsit.
if not `setNext` a root memtracker，the memory usage actually not be detected by process level of TiFlash.
The config `max_memory_usage_for_all_queries` will not work on it.

Reply:
> These are the background threads that handle background tasks such as compaction and delta merge.
> 
> 
> 
> What is the use of `setNext` a root MemoryTracker?

ref：https://pingcap.feishu.cn/wiki/wikcnvA5vz6Mfk9VocSssEqPp3b

Reply:
will it use a lot of memory？ for example：insert or query？

Reply:
> Root memtracker is total_mem_tracker in ProcessLsit.
> if not setNext a root memtracker，the memory usage actually not be detected by process level of TiFlash.
> The config max_memory_usage_for_all_queries will not work on it.

The memory used in here is not caused by queries, so I think it should not be a child of the `ProcessList::total_mem_tracker`(limited by `max_memory_usage_for_all_queries`).
But we do need another root MemoryTracker for all background tasks and maybe a new config item for controlling the memory usage for it.

Reply:
if there are two root memTracker，none of them can track all the memory of the tiflash process. In that case，we can track and throttle the process memory

Reply:
1. I agree that a root memory tracker that track all memory of a TiFlash instance can avoid OOM under some case. But it is weird that config with the name `max_memory_usage_for_all_queries` limit the memory usage for non-queries task. Maybe we need to refactor the config ;)
2. Some background tasks are lower priority than queries, so those background tasks can be delayed or canceled if the total memory usage reaches the high water mark. We need another root memory tracker besides queires_memory_tracker.